### PR TITLE
Leaflet migration

### PR DESCRIPTION
**Pull request by fuentesloic**

### CODE REVIEW

File: static/js/utils.js
Comment (vinyll) [inline static/js/utils.js]: mhhh… why would you call alert with only 1 argument?
What would be the type?

File: static/js/utils.js
Comment (vinyll) [inline static/js/utils.js]: `const notification = document.querySelector('#notification') || document.createElement('div')`
But why would `#notification` not be found? Isn't it supposed to be in the layout?
Notification is not just a plain blank `div`, it has attributes.

**Comment by vinyll** [comment]
And what is this commit about changing alert and notifications doing in leaflet migration??

File: static/js/utils.js
Comment (fuentesloic) [reply static/js/utils.js]: because when an user block the location from his browser, the browser run alert as `alert(message)`
so message is `undefined` and type is a message

File: static/js/utils.js
Comment (fuentesloic) [reply static/js/utils.js]: you write this ... :)

**Comment by fuentesloic** [comment]
It's not false :) I put it here because the probleme is appear when the browser send me `alert()` 
error during localization

File: static/js/utils.js
Comment (vinyll) [reply static/js/utils.js]: Well, you let this go through the review without giving an explicit reason? 😛 
It's either not applicable anymore, or it's missing a comment to explain why I needed to do this :/

File: static/js/utils.js
Comment (vinyll) [reply static/js/utils.js]: Are you saying that the browser is calling the `alert()` function by itsself when rejecting geolocation?
It would then make sense, though probably in another PR (as it's not specific to this PR)

**Comment by vinyll** [comment]
> It's not false :) I put it here because the probleme is appear when the browser send me alert()
error during localization

Ok, I see the point.
Notice that the type would still exist and is used for the _classname_. for an `alert('test me now')`, you would end with a `<div class="test me now">test me now</div>`. We probably want to set a class by default (therefore a `type` which may default to "_info_") so the html does not break.

File: static/js/utils.js
Comment (fuentesloic) [reply static/js/utils.js]: I will cut and paste your comments in the new PR

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: why removing this space?

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: jsut curious, but why setting a class instead of using `:first-of-type` or `:first-child`

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: lowercase?

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: leaflet goes up to 1010 if I remember well. You may want to set it to that level instead.

File: static/css/main.css
Comment (vinyll) [inline static/css/main.css]: why setting a class for the img?

File: static/css/trips-map.css
Comment (vinyll) [inline static/css/trips-map.css]: `0`? That's the browser's default

File: static/css/trips-map.css
Comment (vinyll) [inline static/css/trips-map.css]: lowercase

File: static/js/around-map.js
Comment (vinyll) [inline static/js/around-map.js]: Is it the only way to access the icon? Most of the time, a property that starts with a `_` should not be call as is. Its "private" hence hackish. No big deal though.

File: static/js/trips-map.js
Comment (vinyll) [inline static/js/trips-map.js]: seems like copy and paste. We'll need to refactor this code into one single spot

File: static/js/trips-map.js
Comment (vinyll) [inline static/js/trips-map.js]: is there no method that doesn't start with a `_`?

File: static/js/trips-map.js
Comment (vinyll) [inline static/js/trips-map.js]: single quotes

File: templates/around/map.html
Comment (vinyll) [inline templates/around/map.html]: you very obviously copy and pasted. Code I've never seen before 😮 
You may remove quotes where not required ;)

File: templates/trips/map.html
Comment (vinyll) [inline templates/trips/map.html]: same

**Review by vinyll — CHANGES_REQUESTED** [review]

File: static/js/around-map.js
Comment (fuentesloic) [reply static/js/around-map.js]: I try many many many way without suceed.
the smart should be: `L.marker(poi.location, {icon: icon, id: poi.id})` that work for alt but not for the `id` and I dont know why :s

**Comment by fuentesloic** [comment]
@vinyll could you check this code? for me it's done (for this part)
I refacto the creation map and marker cache (I hope you was thinking to do as I did ^^)

@AymericFaivre @newick : what do you think about the map and color marker ?
Like all know it, I'm not good with this kind of things, we have some suggestion from: 
[mapbox](https://www.mapbox.com/maps/#ready-to-use)
feel you free to propose other source if you want :)
let me know (help me plzzzzz ^^) wich one is better according your designer soul :)

For the marker I have 3 colors:
- blue for a standard marker,
- orange (eurk) when you hover a card (that change the color of its pair marker)
- green when you select the pair poi marker/card

thinks guyss :)

File: static/js/around-map.js
Comment (vinyll) [reply static/js/around-map.js]: Maybe `id` is a reserved key name

File: static/components/PoiCard.js
Comment (vinyll) [inline static/components/PoiCard.js]: remove the line if it's unused

File: static/components/PoiCard.js
Comment (vinyll) [inline static/components/PoiCard.js]: remove if unused

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: space before `{`
Where is the _hovering_ that the comment says?

File: static/js/around-map.js
Comment (vinyll) [inline static/js/around-map.js]: I guess I'd replace the whole `getTwins` function and the forEach function with the simpler:
```js
document.getElementById(poi_id).classList.toggle('current-marker')
document.querySelector(`#card-${poi_id}`).classList.toggle('current-card')
```

**Review by vinyll — CHANGES_REQUESTED** [review]

File: static/components/PoiCard.js
Comment (vinyll) [inline static/components/PoiCard.js]: Why not taking a value like 100m?

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: Are you sure you're not supposed to go up to 1010? (leaflet's top layer)

File: static/js/around-map.js
Comment (vinyll) [inline static/js/around-map.js]: remove

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: space before `!important` (for consistency)

File: static/css/_article_list.css
Comment (vinyll) [inline static/css/_article_list.css:51]: you 😈 `666`

File: static/js/map.js
Comment (vinyll) [inline static/js/map.js:30]: 👍

**Review by vinyll — APPROVED** [review]

File: templates/around/map.html
Comment (vinyll) [inline templates/around/map.html]: `crossorigin=`?
I'm not sure what this does, however it should have quotes `"` for it's empty value, or have no `=` at all!

**Review by vinyll — CHANGES_REQUESTED** [review]

File: static/components/Comments.js
Comment (vinyll) [inline static/components/Comments.js]: single quotes 😛

**Review by vinyll — CHANGES_REQUESTED** [review]

File: static/components/Comments.js
Comment (fuentesloic) [reply static/components/Comments.js]: crap! :)

File: templates/around/map.html
Comment (fuentesloic) [reply templates/around/map.html]: without `=` we have an error

File: static/css/_article_list.css
Comment (fuentesloic) [reply static/css/_article_list.css:51]: mouahahahahah

File: templates/around/map.html
Comment (vinyll) [reply templates/around/map.html]: In which case you could then say `crossorigin=""`.
What's the error message BTW?

**Comment by newick** [comment]
@fuentesloic could we have some screenshots ? :D

**Comment by fuentesloic** [comment]
- blue for a standard marker,
- orange (eurk) when you hover a card (that change the color of its pair marker)
- green when you select the pair poi marker/card

output_jzx4gg

**Comment by newick** [comment]
First of all, add a white border on every card, like that it won't move on hover.
Remember that hover don't work on mobile ;)

I'm ok with the idea, for the color, let's see with @AymericFaivre 😛

**Comment by fuentesloic** [comment]
@newick thinks for it, concerning the mobile version there are not cards, just the map with a popup displayed when the user click on them

**Comment by AymericFaivre** [comment]
Hey !
When you hover a card, maybe you could put all the none selected/hover pin in grey and keep the over pin in blue?

So the 2 pins (selected and hover) would be well localised. 

Is it possible? Or a good idea?

**Comment by vinyll** [comment]
All good for me. I let you deal with the gfx guyz. @newick / @AymericFaivre: I'll let you approve this PR when you agree with @fuentesloic 😗

**Comment by fuentesloic** [comment]
@AymericFaivre : yes sure it's possible, should we modify  the changement on card according the actions ?

**Comment by vinyll** [comment]
Note that clicking twice on the same icon keeps it orange.
I would accuse `toggle`…

screen shot 2017-09-05 at 9 46 16 am

**Comment by AymericFaivre** [comment]
@fuentesloic yep ;)
If it's possible to have a screenshot to be sure it's a good idea it would be perfect :)

**Comment by fuentesloic** [comment]
@AymericFaivre : what do you think ?
output_gjqhvl

**Comment by fuentesloic** [comment]
@vinyll that was the toggle ^^

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: spaces after commas :p

File: static/css/around-map.css
Comment (vinyll) [inline static/css/around-map.css]: could you rephrase this comment?

File: templates/trips/map.html
Comment (vinyll) [inline templates/trips/map.html]: no quotes

File: templates/around/map.html
Comment (vinyll) [inline templates/around/map.html]: no quotes

**Comment by vinyll** [comment]
`toggle` is here again!
clicking on a gray marker makes it green
clicking on a green marker makes it blue
clicking outside of a green marker removes the green one to select the one clicked and make it green.
You end with a bunch of grey markers, some blue markers, and one green.
Nice rainbow. What's the meaning of the blue?

screen shot 2017-09-05 at 7 13 18 pm

**Comment by AymericFaivre** [comment]
I thinks it's good to me @fuentesloic , if we can remove the bug that @vinyll saw ;)

**Comment by fuentesloic** [comment]
@vinyll It's perfectly works on my laptop, could you re build the preact app plz

**Comment by fuentesloic** [comment]
@vinyll : I just forget to change the click action ...
Edit: worse than that I myself create the bug ^^ It's solve with the next commit

File: static/css/around-map.css
Comment (fuentesloic) [reply static/css/around-map.css]: I have to install a plugin to see this fucking invisible spaces (if you have a name ...)

Comment (vinyll) [reply static/css/around-map.css]: I just meant `box-shadow: 0 1rem 2rem 0 rgba(0, 0, 0, 0.2);` ;)

**Comment by fuentesloic** [comment]
I think all is good now, another try required ;)

**Comment by AymericFaivre** [comment]
Thanks !

**Review by vinyll — APPROVED** [review]
Just need to squash most commits to have only a few

**Pull request merged**
